xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-script/src/main/java/org/xwiki/security/authentication/script/AuthenticationScriptService.java

```diff
@@ -121,4 +140,94 @@ public void resetAuthenticationFailureCounter(String username)
     {
         this.authenticationFailureManager.resetAuthenticationFailureCounter(username);
     }
+
+    /**
+     * Compute a relative URL for an {@link AuthenticationResourceReference} based on the given action string.
+     * See {@link AuthenticationResourceReference.AuthenticationAction} for more information.
+     *
+     * @param action the authentication action from which to build the right URL.
+     * @param params the query string parameters of the URL.
+     * @return a relative URL for the current wiki.
+     * @since 13.1RC1
+     */
+    @Unstable
+    public String getAuthenticationURL(String action, Map<String, Object> params)
+    {
+        try {
+            AuthenticationResourceReference.AuthenticationAction authenticationAction =
+                AuthenticationResourceReference.AuthenticationAction.getFromRequestParameter(action);
+
+            AuthenticationResourceReference resourceReference =
+                new AuthenticationResourceReference(authenticationAction);
+            if (params != null) {
+                for (Map.Entry<String, Object> entry : params.entrySet()) {
+                    resourceReference.addParameter(entry.getKey(), entry.getValue());
+                }
+            }
+            ExtendedURL extendedURL = this.defaultResourceReferenceSerializer.serialize(resourceReference);
+            return extendedURL.serialize();
+        } catch (IllegalArgumentException | SerializeResourceReferenceException
+            | UnsupportedResourceReferenceException e) {
+            logger.error("Error while getting authentication URL for action [{}]: [{}]", action,
+                ExceptionUtils.getRootCauseMessage(e));
+            return null;
+        }
+    }
+
+    /**
+     * Request a password reset for the given user.
+     * This will result in computing a verification code and sending the appropriate link by email to the user.
+     * This method returns the email address used, so that we can display it to the user.
+     *
+     * @param user the user for which to perform a reset password request.
+     * @return the email address used to send the verification code.
+     * @throws ResetPasswordException if any error occurs for performing the reset password request.
+     * @since 13.1RC1
+     */
+    @Unstable
+    public InternetAddress requestResetPassword(UserReference user) throws ResetPasswordException
```

Comment:
Shouldn't this method be limited to author with programming right ?

Reply:
Hmmmm good catch, I've been careful to check possible attacks on the other ones but looks like I missed the obvious one...